roxctl/declarativeconfig/create/auth_provider.go

```diff
@@ -0,0 +1,294 @@
+Example of a group: --groups-key "email" --groups-value "my@domain.com" --groups-role "Admin"`)
+	cmd.PersistentFlags().StringSliceVar(&authProviderCmd.groupsRoles, "groups-role", []string{},
+		`role of the groups to add within the auth provider. Note that the tuple of key, value, role should
+be of the same length.
+Example of a group: --groups-key "email" --groups-value "my@domain.com" --groups-role "Admin"`)
+
+	cmd.MarkFlagsRequiredTogether("name", "ui-endpoint")
+
+	cmd.AddCommand(
+		authProviderCmd.oidcCommand(),
+		authProviderCmd.samlCommand(),
+		authProviderCmd.iapCommand(),
+		authProviderCmd.userPKICommand(),
+		authProviderCmd.openShiftCommand(),
+	)
+
+	return cmd
+}
+
+type authProviderCmd struct {
+	authProvider *declarativeconfig.AuthProvider
+
+	requiredAttributes map[string]string
+	claimMapping       map[string]string
+
+	groupsKeys   []string
+	groupsValues []string
+	groupsRoles  []string
+
+	oidcConfig    *declarativeconfig.OIDCConfig
+	samlConfig    *declarativeconfig.SAMLConfig
+	iapConfig     *declarativeconfig.IAPConfig
+	userPKIConfig *declarativeconfig.UserpkiConfig
+
+	// Custom mappings for SAML command.
+	samlIDPCertFile string
+
+	// Custom mapping for User PKI command.
+	userPKICAFile string
+
+	env environment.Environment
+}
+
+func (a *authProviderCmd) oidcCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "oidc",
+		Args:  cobra.NoArgs,
+		RunE:  a.RunE(),
+		Short: "Create a declarative configuration for an OIDC auth provider",
+	}
+	a.oidcConfig = &declarativeconfig.OIDCConfig{}
+
+	cmd.Flags().StringVar(&a.oidcConfig.Issuer, "issuer", "",
+		"issuer of the OIDC client")
+	cmd.Flags().StringVar(&a.oidcConfig.CallbackMode, "mode", "auto",
+		"The callback mode to use. Possible values are: auto, post, query, fragment")
+	cmd.Flags().StringVar(&a.oidcConfig.ClientID, "client-id", "",
+		"Client ID of the OIDC client")
+	cmd.Flags().StringVar(&a.oidcConfig.ClientSecret, "client-secret", "",
+		"Client Secret of the OIDC client")
+	cmd.Flags().BoolVar(&a.oidcConfig.DisableOfflineAccessScope, "disable-offline-access", false,
+		"Disable requesting the scope offline_access from the OIDC identity provider. This should only be set "+
+			"if there are any limitations from your OIDC IdP about the number of sessions with the offline_access scope")
+
+	cmd.Flags().StringToStringVar(&a.claimMapping, "claim-mappings", map[string]string{},
+		`list of non-standard claims from the IdP token that should be available within auth provider rules, e.g.
+--claim-mappings "my_claim_on_the_idp_token=claim_name_on_the_rox_token"`)
+
+	return cmd
+}
+
+func (a *authProviderCmd) samlCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "saml",
+		Args:  cobra.NoArgs,
+		RunE:  a.RunE(),
+		Short: "Create a declarative configuration for a SAML auth provider",
+	}
+	a.samlConfig = &declarativeconfig.SAMLConfig{}
+
+	cmd.Flags().StringVar(&a.samlConfig.SpIssuer, "sp-issuer", "", "service provider "+
```

Comment:
Can we mark `sp-issuer` as required? Also `metadata-url` and other 4 parameters as mutually exclusive, and other 4 parameters as required together?